byway/src/main/java/com/google/sps/servlets/DestinationsServlet.java

```diff
@@ -24,62 +26,53 @@
 
   private final Gson gson = new Gson();
   private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-  private Key userKey;
-
-  @Override
-  public void init() {
-    Entity userEntity = new Entity("UserInputs");
-    userEntity.setProperty("start", "");
-    userEntity.setProperty("destinations", new ArrayList<String>());
-    datastore.put(userEntity);
-    userKey = userEntity.getKey();
-  }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Key tripKey = KeyFactory.stringToKey(request.getParameter("tripKey"));
     Entity entity;
     try {
-      entity = datastore.get(userKey);
+      entity = datastore.get(tripKey);
     } catch (EntityNotFoundException e) {
-      UserLocations userLocations = new UserLocations("", new ArrayList<String>());
-      response.setContentType("application/json;");
-      response.getWriter().println(gson.toJson(userLocations));
-      logger.atInfo().withCause(e).log("Unable to find UserLocations Entity %s", userKey);
+      logger.atInfo().withCause(e).log("Unable to find Trip Entity %s", tripKey);
+      response.setStatus(HttpServletResponse.SC_NOT_FOUND);
       return;
     }
     String start = (String) entity.getProperty("start");
     ArrayList<String> destinations = (ArrayList<String>) entity.getProperty("destinations");
```

Comment:
You have a special class for representing `Trips`. It has a special method for pulling out the various properties of an `entity` which represents a trip. Use that! Every additional place where you directly access the string properties of the trip `Entity` is another place where you could introduce a bug, and another place you have to *remember* to change, if you modify what properties the trip entity actually has. Note this difference: if you typo `destinatins`, you won't find out until you run your code. If you decide to rename `destinations` to `pointsOfInterest` or something, you have to *remember* to *manually* update all places you wrote that string. If you access `Trip.getDestinatins()`, you'll find out about that typo at compile time, because that method won't exist. Same if you rename something. In general, try to access things by string-key in as few places as possible, and try to do all those accesses in one class, so that if you have to change them, they're all in one place to fix. Also consider making constants for the string keys if you have to use them in more than ~2 places.

Reply:
fixed